llama-index-integrations/postprocessor/llama-index-postprocessor-rankllm-rerank/llama_index/postprocessor/rankllm_rerank/base.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, List, Optional
+
+from llama_index.core.bridge.pydantic import Field, PrivateAttr
+from llama_index.core.postprocessor.types import BaseNodePostprocessor
+from llama_index.core.schema import NodeWithScore, QueryBundle
+
+
+class RankLLMRerank(BaseNodePostprocessor):
+    """RankLLM-based reranker."""
+    top_n: int = Field(default=5, description="Top N nodes to return from reranking.")
+    model: str = Field(default="zephyr", description="Reranker model name.")
+    with_retrieval: bool = Field(default=False, description="Perform retrieval before reranking.")
+    _model: Any = PrivateAttr()
+    _result: Any = PrivateAttr()
+    _retriever: Any = PrivateAttr()
+    
+    def __init__(
+        self,
+        top_n: int = 10,
+        with_retrieval: bool = False,
+        model: str = "zephyr",
+    ):
+        try: 
+            from rank_llm.result import Result 
+            self._result = Result
+
+
+            if model == "vicuna":
+                from rank_llm.rerank.vicuna_reranker import VicunaReranker
+                self._model = VicunaReranker()
+            else:
+                from rank_llm.rerank.zephyr_reranker import ZephyrReranker
+                self._model = ZephyrReranker()
+                
+            if with_retrieval:
+                from rank_llm.retrieve.retriever import Retriever
+                self._retriever = Retriever
+
+        except ImportError:
+            raise ImportError(
+                "Cannot import rank_llm",
+                "please `pip install rank_llm`",
+            )
+
+        super().__init__(
+            top_n=top_n,
+            with_retrieval=with_retrieval,
+            model=model,
+        )
+    
+    @classmethod
+    def class_name(cls) -> str:
+        return "RankLLMRerank"
+    
+    def _postprocess_nodes(
+        self,
+        nodes: List[NodeWithScore],
+        query_bundle: Optional[QueryBundle] = None,
+    ) -> List[NodeWithScore]:
+        if query_bundle is None:
+            raise ValueError("Query bundle must be provided.")
```

Comment:
I think this means we probably shouldn't make `query_bundle` an Optional type?

Reply:
Sure